File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int i = 0,j;
        int maxi = 0;
        unordered_map<char,int> mp;
        for (int j = 0;j < s.length(); j++){
            mp[s[j]]++;
            while(mp[s[j]]>1){
                mp[s[i++]]--;

            }
            maxi = max(maxi,j-i+1);
        }
        return maxi;
        }
};

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        i = 0
        dic = dict()
        l = 0 
        for j in range(len(s)):

            if s[j] not in dic:
                dic[s[j]]=1 
            else:
                dic[s[j]]+=1 
            
            while dic[s[j]] > 1:
                dic[s[i]]-=1 
                i+=1
            
            l = max(l, j-i+1)

        return l

